### DISCUSSION TITLE
[FEATURE] Lago integration for subscription management.

### DISCUSSION THREAD

**Discussion by zeluspudding** [juspay/hyperswitch#2821] [Ideas / Feature Requests]
### Feature Description

I've made a similar feature request over at [Lago](https://getlago.canny.io/feature-requests/p/add-hyperswitch-integration). The paring of these two open source project seems very natural and symbiotic. Hyperswitch is missing subscription management and Lago is missing most payment provider integrations.

### Possible Implementation

Lago is opensource, so pulling it into this project, or joining forces with their contributors would be a natural starting point.

### Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

Potentially

**Comment by manojradhakrishnan** [comment]
@zeluspudding Thanks for adding this issue.

Could you please propose a high level solution approach on how we can achieve this feat together.
